src/main/java/com/questhelper/quests/tribaltotem/CombinationLockStep.java

```diff
@@ -0,0 +1,123 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.questhelper.quests.tribaltotem;
+
+import com.questhelper.QuestHelperPlugin;
+import com.questhelper.questhelpers.QuestHelper;
+import com.questhelper.steps.QuestStep;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.widgets.Widget;
+import net.runelite.client.eventbus.Subscribe;
+
+import java.awt.*;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CombinationLockStep extends QuestStep {
+
+    private final String ENTRY_ONE = "K";
+    private final String ENTRY_TWO = "U";
+    private final String ENTRY_THREE = "R";
+    private final String ENTRY_FOUR = "T";
+
+    private final int SLOT_ONE = 42;
+    private final int SLOT_TWO = 43;
+    private final int SLOT_THREE = 44;
+    private final int SLOT_FOUR = 45;
+
+    private final int ARROW_ONE_LEFT = 46;
+    private final int ARROW_ONE_RIGHT = 47;
+    private final int ARROW_TWO_LEFT = 48;
+    private final int ARROW_TWO_RIGHT = 49;
+    private final int ARROW_THREE_LEFT = 50;
+    private final int ARROW_THREE_RIGHT = 51;
+    private final int ARROW_FOUR_LEFT = 52;
+    private final int ARROW_FOUR_RIGHT = 53;
+
+    private final int COMPLETE = 55;
+
+    private final HashMap<Integer, Integer> highlightButtons = new HashMap<>();
+
+    public CombinationLockStep(QuestHelper questHelper) {
+        super(questHelper, "Click the highlighted arrows to move the slots to the correct letters.");
+        highlightButtons.put(1, ARROW_ONE_RIGHT);
+        highlightButtons.put(2, ARROW_TWO_RIGHT);
+        highlightButtons.put(3, ARROW_THREE_RIGHT);
+        highlightButtons.put(4, ARROW_FOUR_RIGHT);
+    }
+
+    @Subscribe
+    public void onGameTick(GameTick gameTick)
+    {
+        updateSolvedPositionState();
+    }
+
+    private void updateSolvedPositionState()
+    {
+        highlightButtons.replace(1, matchStateToSolution(SLOT_ONE, ENTRY_ONE, ARROW_ONE_RIGHT, ARROW_ONE_LEFT));
+        highlightButtons.replace(2, matchStateToSolution(SLOT_TWO, ENTRY_TWO, ARROW_TWO_RIGHT, ARROW_TWO_LEFT));
+        highlightButtons.replace(3, matchStateToSolution(SLOT_THREE, ENTRY_THREE, ARROW_THREE_RIGHT, ARROW_THREE_LEFT));
+        highlightButtons.replace(4, matchStateToSolution(SLOT_FOUR, ENTRY_FOUR,ARROW_FOUR_RIGHT, ARROW_FOUR_LEFT));
+        if (highlightButtons.get(1) + highlightButtons.get(2) + highlightButtons.get(3) + highlightButtons.get(4) == 0)
+        {
+            highlightButtons.put(5, COMPLETE);
+        } else {
+            highlightButtons.put(5, 0);
+        }
+    }
+
+    private int matchStateToSolution(int slot, String target, int arrowRightId, int arrowLeftId)
+    {
+        Widget widget = client.getWidget(369, slot);
+        if (widget == null) return 0;
+        String current = widget.getText();
+        int Id = current.charAt(0) < target.charAt(0) ? arrowRightId : arrowLeftId;
```

Comment:
You could make this choose to go backwards/forwards considering proximity in both directions by doing something like the following:

```
char current = widget.getText().charAt(0);
int currentPos = (int)current - (int)'A';
int id = Math.floorMod(currentPos - target, 26) < Math.floorMod(target - currentPos, 26) ? arrowRightId : arrowLeftId;
```